daemon/state.go

```diff
@@ -70,10 +70,13 @@ func (d *Daemon) validateEndpoint(ep *endpoint.Endpoint) (valid bool, err error)
 	}
 
 	if ep.K8sPodName != "" && ep.K8sNamespace != "" && k8s.IsEnabled() {
-		_, err := k8s.Client().CoreV1().Pods(ep.K8sNamespace).Get(ep.K8sPodName, meta_v1.GetOptions{})
+		p, err := k8s.Client().CoreV1().Pods(ep.K8sNamespace).Get(ep.K8sPodName, meta_v1.GetOptions{})
 		if err != nil && k8serrors.IsNotFound(err) {
 			return false, fmt.Errorf("kubernetes pod not found")
 		}
+		if err == nil {
+			ep.UpdateVisibilityPolicy(p.Annotations)
```

Comment:
The restore logic handles regeneration in quite a special way; regeneration is triggered here: https://github.com/cilium/cilium/blob/75c77f1c1c273091d26af31a1485fc0090a181ec/daemon/state.go#L200 .

Agreed re: code organization. Will update the comment. I wanted to piggy-back on top of existing logic since we already are querying the API here.

Reply:
OK, I thought it was to do with the API query here. Might also be useful to leave a comment in the code pointing out how the regeneration works. Up to you.